src/data/list/basic.lean

```diff
@@ -4215,7 +4215,7 @@ begin
   { rw [eq_nil_of_le hml, filter_ge_of_top_le hml] }
 end
 
-@[simp] lemma filter_ge (n m l : ℕ) : (Ico n m).filter (λ x, x ≥ l) = Ico (max n l) m :=
+@[simp] lemma filter_ge (n m l : ℕ) : (Ico n m).filter (λ x, x ≥ l) = Ico (_root_.max n l) m :=
```

Comment:
please make `list.max` protected

Reply:
What about defining `list.max` this way `def max (l : list α) : α := l.foldr max l.head`

Reply:
That should be for `imax`. I also like this definition because we have laws about `foldl / foldr` but I would advocate using `foldl` because it's tail recursive.

Reply:
Good point. I'll make it protected and use `foldl` in the next commit